### ISSUE TITLE
Stuck at "Gathering uploaded artists from YouTube Music..." for 8 hours.

### ISSUE THREAD

**Issue by asktoby** [jamesbrindle/YTMusicUploader#10]
The software had been working well (19k out of 33k files uploaded) but my PC crashed.
Now when I launch it it has been stuck at "Gathering uploaded artists from YouTube Music..." for 8 hours.
Is there any way to "kick start" it?

**Comment by asktoby** [comment]
I've found and clicked on "Reset uploaded music states" inside the cog menu and that seems to have done the trick.

**Comment by phuqertrfc** [comment]
I am also having this same issue.
I transferred from Play Music a while back but noticed not everything had transferred, plus I didn't like the fact I had to manually add new music.
So I've installed this app and ran it for the first time.
It discovered 34801 files and started to go through them all. Most of them showed as already present but a few that were missing got added.
It was running really well but then when it got to 20532 uploaded it just stopped and hung at "Gathering uploaded artists"
I left it for hours but nothing.
I restarted but just hung again.
I clicked on "Reset uploaded music states" and it started from 0 again, but then it hung at exactly 20532 uploads for the second time.
I did wonder if it was an issue with the remaining files, but the second time I ran it, it processed different files as it was still finding stuff that hadn't been uploaded, so it seems to be an issue with this tool rather than the files.
Seems strange that it hangs at exactly the same number each time though.
I tried a third time and exactly the same result.

I also tried adding some new music to the folder to see if it picked that up and the discovered files increased but the uploads still hung at 20532.

Any ideas how to get around this?

**Comment by asktoby** [comment]
I've had some success getting it unstuck by removing any items from the upload queue and then adding them again but a little bit different, e.g. adding them from one folder up, or mapping them as a fresh letter, so to the software thinks it's a whole new folder.
Then it seems to quickly checksum most of the files and upload any remaining ones.
I also found one file that was fouling the upload, it was a 1 hour mix. I just took that out of the folder to stop it sticking. However in that case I could see the filename listed in the software so it was obvious to me that it was the culprit.

**Comment by jamesbrindle** [comment]
Just to let you know... I was supposed to look at fixing this last weekend, but ended up working all weekend. I've made a good number of changes though tonight, which could address the issue. If not... I've introduced some decent logging, so if the problem persists I'm hopeful I can get to the bottom of it. I haven't deployed the changes yet... I've got a couple more things to do and I'll get this pushed tomorrow afternoon.
